src/routes/SafeAppLandingPage/components/UserSafeSection.tsx

```diff
@@ -16,35 +17,65 @@ import UseYourSafe from './UseYourSafe'
 type UserSafeProps = {
   safeAppUrl: string
   availableChains: string[]
-  safeAppChainId: string | null
+  safeAppChainId: string
+}
+
+const getCompatibleSafes = (
+  safesFromLocalStorage: LocalSafes,
+  safesFromService: Record<string, string[]>,
+  compatibleChains: string[],
+  addressBook: AddressBookEntry[],
+): AddressBookEntry[] => {
+  return compatibleChains.reduce((result, chainId) => {
+    const flatSafesFromLocalStorage = safesFromLocalStorage[chainId]?.map(({ address }) => address) || []
```

Comment:
perhaps can be removed if you use https://lodash.com/docs/#uniqBy with an iteratee `x => x.address || x`

Reply:
I tried to use `uniqBy`, but it returns the complete element (no map is performed).
In our case, if we use `uniqBy`, it returns a `(string | SafeRecordProps)[]` and we need only a `string[]`  (so at the end we need to manually do a mapping to get `string[]`)